node-src/tasks/gitInfo.ts

```diff
@@ -59,11 +59,19 @@ export const setGitInfo = async (ctx: Context, task: Task) => {
     patchBaseRef,
     fromCI: ci,
     interactive,
+    isLocalBuild,
   } = ctx.options;
 
+  const commitAndBranchInfo = await getCommitAndBranch(ctx, { branchName, patchBaseRef, ci });
+
+  const creatorEmail = isLocalBuild
+    ? await getUserEmail()
+    : commitAndBranchInfo.committerEmail || 'unknown';
```

Comment:
I don't think we should record the committer as the creator for non-local builds. It would make it harder to tell where the data originated (the commit or the git config) whilst not providing any benefit (we already record `committerEmail` so it would be duplicative).

Reply:
We should consider hashing the email and recording `creatorId` or something like that instead, in order to avoid recording (even more) PII. We don't need the actual plaintext email, right? We can always hash the `committerEmail` as well if we want to cross-reference a CI build with a local build. Gravatar's [implementation](https://en.gravatar.com/site/implement/hash) (md5) seems reasonable.

Reply:
What happens if (for whatever reason) we can't retrieve the Git user email when running a local build? I think it should not be allowed to do that because we wouldn't be able to track local baselines.

Reply:
> What's the purpose of recording unknown? Usually we do that only when retrieving data, not when recording it.

I guess we don't need to *record* `'unknown'`, as really the `creatorEmail` doesn't have any effect on non-local builds anyway (at the moment).

The purpose for it on non-local builds is in baseline selection (ie. queries we do). In these cases we want to pass *something* as `onlyLocalBuildsFrom` on CI builds if we can't calculate the commit's author. That will have the effect of filtering out *all* local builds, which I think is better than including all local builds.

We probably also need to show a large warning when this happens.

> Let's just always record the configured user email (preferably hashed), even for CI builds. We already track the committerEmail, so it would be duplicative anyway.

Given the above, we might consider not even storing a `creatorEmail` on CI builds? I could go either way.

> We should consider hashing the email and recording creatorId or something like that instead, in order to avoid recording (even more) PII. 

OK, sure, let's start hashing (the new) email fields.

> What happens if (for whatever reason) we can't retrieve the Git user email when running a local build? I think it should abort the build (throw an error) because we wouldn't be able to track local baselines. Falling back to unknown is likely to cause weird issues, but the above logic isn't doing that (for local builds) anyway.

I think it would throw an error but I'll verify that.